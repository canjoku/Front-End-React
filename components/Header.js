import { Component } from 'react'
import Navbar from './Navbar'

class Header extends Component {
  render () {
    const { mutation } = this.props
    return (
      <header className="c-header">
        <div className="c-header__content">
          <h1 className="c-header__title">TBlog Stories</h1>
          <div className="c-header_subtitle">
            <div className="c-header__description">
              A place to read about web development code stories. Courtesy of <a href="#">Tony Njoku.</a>
              <Navbar mutation={mutation}/>
            </div>
          </div>
        </div>
      </header>
    )
  }
}

export default Header
