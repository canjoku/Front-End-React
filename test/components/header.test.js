import Header from '../../components/Header'


test('<Header /> renders correctly', () => {
  const component = shallow(<Header />)
  expect(component).toMatchSnapshot()
})
