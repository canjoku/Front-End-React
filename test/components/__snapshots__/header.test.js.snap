// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="c-header__content"
      >
        <h1
          className="c-header__title"
        >
          TBlog Stories
        </h1>
        <div
          className="c-header_subtitle"
        >
          <div
            className="c-header__description"
          >
            A place to read about web development code stories. Courtesy of 
            <a
              href="#"
            >
              Tony Njoku.
            </a>
            <Navbar
              mutation={undefined}
            />
          </div>
        </div>
      </div>,
      "className": "c-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="c-header__title"
          >
            TBlog Stories
          </h1>,
          <div
            className="c-header_subtitle"
          >
            <div
              className="c-header__description"
            >
              A place to read about web development code stories. Courtesy of 
              <a
                href="#"
              >
                Tony Njoku.
              </a>
              <Navbar
                mutation={undefined}
              />
            </div>
          </div>,
        ],
        "className": "c-header__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TBlog Stories",
            "className": "c-header__title",
          },
          "ref": null,
          "rendered": "TBlog Stories",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="c-header__description"
            >
              A place to read about web development code stories. Courtesy of 
              <a
                href="#"
              >
                Tony Njoku.
              </a>
              <Navbar
                mutation={undefined}
              />
            </div>,
            "className": "c-header_subtitle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "A place to read about web development code stories. Courtesy of ",
                <a
                  href="#"
                >
                  Tony Njoku.
                </a>,
                <Navbar
                  mutation={undefined}
                />,
              ],
              "className": "c-header__description",
            },
            "ref": null,
            "rendered": Array [
              "A place to read about web development code stories. Courtesy of ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tony Njoku.",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Tony Njoku.",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "mutation": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="c-header__content"
        >
          <h1
            className="c-header__title"
          >
            TBlog Stories
          </h1>
          <div
            className="c-header_subtitle"
          >
            <div
              className="c-header__description"
            >
              A place to read about web development code stories. Courtesy of 
              <a
                href="#"
              >
                Tony Njoku.
              </a>
              <Navbar
                mutation={undefined}
              />
            </div>
          </div>
        </div>,
        "className": "c-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="c-header__title"
            >
              TBlog Stories
            </h1>,
            <div
              className="c-header_subtitle"
            >
              <div
                className="c-header__description"
              >
                A place to read about web development code stories. Courtesy of 
                <a
                  href="#"
                >
                  Tony Njoku.
                </a>
                <Navbar
                  mutation={undefined}
                />
              </div>
            </div>,
          ],
          "className": "c-header__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "TBlog Stories",
              "className": "c-header__title",
            },
            "ref": null,
            "rendered": "TBlog Stories",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="c-header__description"
              >
                A place to read about web development code stories. Courtesy of 
                <a
                  href="#"
                >
                  Tony Njoku.
                </a>
                <Navbar
                  mutation={undefined}
                />
              </div>,
              "className": "c-header_subtitle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "A place to read about web development code stories. Courtesy of ",
                  <a
                    href="#"
                  >
                    Tony Njoku.
                  </a>,
                  <Navbar
                    mutation={undefined}
                  />,
                ],
                "className": "c-header__description",
              },
              "ref": null,
              "rendered": Array [
                "A place to read about web development code stories. Courtesy of ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tony Njoku.",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Tony Njoku.",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "mutation": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
